nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/AICorpus.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.cloud;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AICorpus {
+
+    protected static final String entity = "document";
+
+    protected static final String type = "AI_Corpus";
+
+    @JsonProperty("name")
+    protected String name;
+
+    @JsonProperty("properties")
+    protected Properties props;
+
+    public AICorpus(@JsonProperty("name") String name, @JsonProperty("properties") Properties props) {
+        this.name = name;
+        this.props = props;
+    }
+
+    @JsonProperty("entity-type")
+    public String getEntity() {
+        return entity;
+    }
+
+    @JsonProperty("type")
+    public String getType() {
+        return type;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Properties getProps() {
+        return props;
+    }
+
+    public void setProps(Properties props) {
+        this.props = props;
+    }
+
+    public static class Properties {
+
+        @JsonProperty("dc:title")
+        protected String title;
+
+        @JsonProperty("ai_corpus:documents_count")
+        protected long docCount;
+
+        @JsonProperty("ai_corpus:evaluation_documents_count")
+        protected long evaluationDocCount;
+
+        @JsonProperty("query")
+        protected String query;
+
+        @JsonProperty("split")
+        protected int split;
+
+        @JsonProperty("fields")
+        protected List<Map<String, Object>> fields;
```

Comment:
Why `Object` as a type in `Map<String, Object>`
The filed should be a field name and type, being type a enum or a static String.